src/java.base/share/classes/jdk/internal/classfile/constantpool/ConstantPoolBuilder.java

```diff
@@ -0,0 +1,561 @@
+     * otherwise, a new entry is added and the new entry is returned.
+     *
+     * @param desc the symbolic descriptor for the class
+     */
+    default Utf8Entry utf8Entry(ClassDesc desc) {
+        return utf8Entry(desc.descriptorString());
+    }
+
+    /**
+     * {@return A {@link Utf8Entry} describing the method descriptor of the provided
+     * {@linkplain MethodTypeDesc}}
+     * If a UTF8 entry in the pool already describes this field descriptor, it is returned;
+     * otherwise, a new entry is added and the new entry is returned.
+     *
+     * @param desc the symbolic descriptor for the method type
+     */
+    default Utf8Entry utf8Entry(MethodTypeDesc desc) {
+        return utf8Entry(desc.descriptorString());
+    }
+
+    /**
+     * {@return A {@link ClassEntry} describing the class whose internal name
+     * is encoded in the provided {@linkplain Utf8Entry}}
+     * If a Class entry in the pool already describes this class,
+     * it is returned; otherwise, a new entry is added and the new entry is
+     * returned.
+     *
+     * @param ne the constant pool entry describing the internal name of the class
+     */
+    ClassEntry classEntry(Utf8Entry ne);
+
+    /**
+     * {@return A {@link ClassEntry} describing the class described by
+     * provided {@linkplain ClassDesc}}
+     * If a Class entry in the pool already describes this class,
+     * it is returned; otherwise, a new entry is added and the new entry is
+     * returned.
+     *
+     * @param classDesc the symbolic descriptor for the class
+     */
+    default ClassEntry classEntry(ClassDesc classDesc) {
+        return classEntry(utf8Entry(Util.toInternalName(classDesc)));
+    }
+
+    /**
+     * {@return A {@link PackageEntry} describing the class whose internal name
+     * is encoded in the provided {@linkplain Utf8Entry}}
+     * If a Package entry in the pool already describes this class,
+     * it is returned; otherwise, a new entry is added and the new entry is
+     * returned.
+     *
+     * @param nameEntry the constant pool entry describing the internal name of
+     *                  the package
+     */
+    PackageEntry packageEntry(Utf8Entry nameEntry);
+
+    /**
+     * {@return A {@link PackageEntry} describing the class described by
+     * provided {@linkplain PackageDesc}}
+     * If a Package entry in the pool already describes this class,
+     * it is returned; otherwise, a new entry is added and the new entry is
+     * returned.
+     *
+     * @param packageDesc the symbolic descriptor for the class
+     */
+    default PackageEntry packageEntry(PackageDesc packageDesc) {
+        return packageEntry(utf8Entry(packageDesc.packageInternalName()));
+    }
+
+    /**
+     * {@return A {@link ModuleEntry} describing the module whose name
+     * is encoded in the provided {@linkplain Utf8Entry}}
+     * If a Module entry in the pool already describes this class,
```

Comment:
(Here and elsewhere) - Module is capitalized. Either you use a lower case name, or you use a capital name, to refer to `ModuleEntry`, or `CONSTANT_Module_info` - e.g. a standalone `Module` with capital `M` is not a concept in this API. (personally I think lower case is just fine).

Reply:
Will fix it, thanks.